drivers/volume/portworx/portworx.go

```diff
@@ -2461,29 +2467,37 @@ func (d *portworx) DecommissionNode(n *node.Node) error {
 		}
 	}
 
-	if err := d.StopDriver([]node.Node{*n}, false, nil); err != nil {
+	if err := d.EnterMaintenance(*n); err != nil {
 		return &ErrFailedToDecommissionNode{
 			Node:  n.Name,
 			Cause: fmt.Sprintf("Failed to stop driver on node: %v. Err: %v", n.Name, err),
 		}
 	}
 
+	time.Sleep(10 * time.Second)
+
 	nodeResp, err := d.getNodeManager().Inspect(d.getContext(), &api.SdkNodeInspectRequest{NodeId: n.VolDriverNodeID})
 	if err != nil {
 		return &ErrFailedToDecommissionNode{
 			Node:  n.Name,
 			Cause: fmt.Sprintf("Failed to inspect node: %v. Err: %v", nodeResp.Node, err),
 		}
 	}
+	logrus.Infof("removing node %v", nodeResp.Node.Id)
 
 	// TODO replace when sdk supports node removal
-	if err = d.legacyClusterManager.Remove([]api.Node{{Id: nodeResp.Node.Id}}, false); err != nil {
-		return &ErrFailedToDecommissionNode{
-			Node:  n.Name,
-			Cause: err.Error(),
+	if err = d.legacyClusterManager.Remove([]api.Node{{Id: nodeResp.Node.Id}}, true); err != nil {
+		if !strings.Contains(err.Error(), "Node remove is pending") {
+			return &ErrFailedToDecommissionNode{
+				Node:  n.Name,
+				Cause: err.Error(),
+			}
 		}
 	}
 
+	logrus.Infof(" %v: Node remove is pending. Waiting for it to complete", nodeResp.Node.Id)
+	time.Sleep(30 * time.Second)
```

Comment:
Sleep need to be replaced with wait and check method ...

Reply:
fixed it